lib/shelley/test/integration/Main.hs

```diff
@@ -152,49 +155,55 @@ import qualified Test.Integration.Scenario.CLI.Shelley.Transactions as Transacti
-                    ByronCoinSelections.spec @n
-                    Wallets.spec @n
-                    ByronWallets.spec @n
-                    HWWallets.spec @n
-                    Migrations.spec @n
-                    ByronMigrations.spec @n
-                    Transactions.spec @n
-                    Network.spec
-                    Network_.spec
-                    StakePools.spec @n
-                    ByronTransactions.spec @n
-                    ByronHWWallets.spec @n
+main = withTestsSetup $ \testDir tracers -> hspec $ do
+    describe "No backend required" $
+        parallel' $ describe "Miscellaneous CLI tests"
+            MiscellaneousCLI.spec
 
-                -- possible conflict with StakePools
-                Settings.spec @n
+    testEras $ \era ->
+        specWithServer (testDir </> clusterEraName era) tracers (Just era) $ do
+            parallel $ describe "API Specifications" $ do
+                Addresses.spec @n
+                CoinSelections.spec @n
+                ByronAddresses.spec @n
+                ByronCoinSelections.spec @n
+                Wallets.spec @n
+                ByronWallets.spec @n
+                HWWallets.spec @n
+                Migrations.spec @n
+                ByronMigrations.spec @n
+                Transactions.spec @n
+                Network.spec
+                Network_.spec
+                StakePools.spec @n
+                ByronTransactions.spec @n
+                ByronHWWallets.spec @n
 
-            -- Hydra runs tests with code coverage enabled. CLI tests run
-            -- multiple processes. These processes can try to write to the
-            -- same .tix file simultaneously, causing errors.
-            --
-            -- Because of this, don't run the CLI tests in parallel in hydra.
-            parallelIf (not nix) $ describe "CLI Specifications" $ do
+                -- Possible conflict with StakePools - mark as not parallizable
+                sequential $ Settings.spec @n
+
+            parallel' $ describe "CLI Specifications" $ do
                 AddressesCLI.spec @n
                 TransactionsCLI.spec @n
                 WalletsCLI.spec @n
                 HWWalletsCLI.spec @n
                 PortCLI.spec
                 NetworkCLI.spec
   where
-    parallelIf :: forall a. Bool -> SpecWith a -> SpecWith a
-    parallelIf flag = if flag then parallel else id
+    -- Hydra runs tests with code coverage enabled. CLI tests run
+    -- multiple processes. These processes can try to write to the
+    -- same .tix file simultaneously, causing errors.
+    --
+    -- Because of this, don't run the CLI tests in parallel in hydra.
+    parallel' :: forall a. SpecWith a -> SpecWith a
+    parallel' spec = do
+        nix <- runIO inNixBuild
+        parallelIf (not nix) spec
+
+    parallelIf flag = if flag then parallel else sequential
+
+    testEras run = forM_ [MaryHardFork, AllegraHardFork] $
+        \era -> describe (show era) $ run era
```

Comment:
We are running the test suite twice, in both Mary and Allegra ?

Reply:
Yes - slower - but I think it's better to be on the safe side.